### PR TITLE
Add TEST_RUN_UNTIL_FAILURE option to integration suite

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Adds the `TEST_RUN_UNTIL_FAILURE=true` option to the integration testing suite.

## Why is it important?

When debugging flaky tests is helpful to have the test run over and over again until it fails.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

### CODE REVIEW

**Comment by pierrehilbert** [comment]
I like this new idea!

File: docs/test-framework-dev-guide.md [hunk 1]
```diff
@@ -116,6 +116,13 @@ using `AGENT_KEEP_INSTALLED=true` environment variable. It is recommend to only
 
 - `AGENT_KEEP_INSTALLED=true mage integration:single [testName]`
 
+#### Run until failure
+In the case that your tracking down a flaky test it is helpful to have it keep running until it fails. The testing
```
Comment (AndersonQ) [inline docs/test-framework-dev-guide.md:120]: typo
```suggestion
In the case that you're tracking down a flaky test it is helpful to have it keep running until it fails. The testing
```

File: magefile.go [hunk 2]
```diff
@@ -2085,3 +2097,12 @@ func hasSnapshotEnv() bool {
...
 	return b
 }
+
+func hasRunUntilFailure() bool {
+	runUntil := os.Getenv("TEST_RUN_UNTIL_FAILURE")
+	if runUntil == "" {
+		return false
+	}
+	b, _ := strconv.ParseBool(runUntil)
+	return b
```
Comment (AndersonQ) [inline magefile.go:2107]: [Suggestion]
As you're ignoring the error, you don't need to test for empty string. It'll fail and return false anyway

```suggestion
	b, _ := strconv.ParseBool(runUntil)
	return b
```

**Review by AndersonQ — APPROVED** [review]

**Comment by blakerouse** [comment]
Buildkite failed due to quota issue in GCP. Re-trying hoping that it was just to many jobs running at once.

**Comment by blakerouse** [comment]
buildkite test this
